From olteanv@gmail.com  Thu Jul 30 09:29:18 2020
From: Vladimir Oltean <olteanv@gmail.com>
Date: Wed, 24 Jun 2020 15:45:17 +0300
Subject: Revert "dpaa_eth: fix usage as DSA master, try 3"
To: stable@vger.kernel.org, gregkh@linuxfoundation.org, netdev@vger.kernel.org
Cc: madalin.bucur@oss.nxp.com, camelia.groza@nxp.com, joakim.tjernlund@infinera.com, fido_max@inbox.ru, linux-kernel@vger.kernel.org
Message-ID: <20200624124517.3212326-1-olteanv@gmail.com>


From: Vladimir Oltean <vladimir.oltean@nxp.com>

This reverts commit 40a904b1c2e57b22dd002dfce73688871cb0bac8.

The patch is not wrong, but the Fixes: tag is. It should have been:

	Fixes: 060ad66f9795 ("dpaa_eth: change DMA device")

which means that it's fixing a commit which was introduced in:

git tag --contains 060ad66f97954
v5.5

which then means it should have not been backported to linux-5.4.y,
where things _were_ working and now they're not.

Reported-by: Joakim Tjernlund <joakim.tjernlund@infinera.com>
Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
Changes in v1:
Adjusted the commit message from linux-4.19.y to linux-5.4.y

Changes in v2:
Fixed the sha1sum of the reverted commit.

 drivers/net/ethernet/freescale/dpaa/dpaa_eth.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/freescale/dpaa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpaa/dpaa_eth.c
@@ -2802,7 +2802,7 @@ static int dpaa_eth_probe(struct platfor
 	}
 
 	/* Do this here, so we can be verbose early */
-	SET_NETDEV_DEV(net_dev, dev->parent);
+	SET_NETDEV_DEV(net_dev, dev);
 	dev_set_drvdata(dev, net_dev);
 
 	priv = netdev_priv(net_dev);
